Linear Event chain for Demon Hunters

FelModder opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added Draenei as a race that can take the decision to become demon hunters
- Added a Spectral Vision trait that replaces the blind trait, so that many interactions and events don't get blocked off just because you are blind (Spectral Vision should still make you get the blindfold)
- Taking the decision to become a demon hunter instead now launches an event chain (1 day after the faith conversion event)
- `demon_hunter_events.0001` and `demon_hunter_events.0002` just describe what happens during the ritual. 0002 will make you incapable and blind for 30 days, as it says in rituals that you go into coma for a while before waking up with properly infused powers of a demon. (you should loose the incapable and blind trait after the 30 days in `demon_hunter_events.0003`)
- `demon_hunter_events.0003` has a chance to give you the possessed trait which itself launches another event in which the demon will threaten you. (if you do not get possessed then that's right where the story for Demon Hunter events ends.)
- `demon_hunter_events.0005` will launch depending on your prowess, if you get the lunatic trait, the event will launch in 1/2 years or the event will launch in 4/5 years if you don't get the lunatic trait.
- When `demon_hunter_events.0005` launches, you will get access to the decision to attain the metamorphosis form.
- Metamorphosis is a trait you get after taking the decision to "Embrace the Demon".
- How to leave the metamorphosis form? Every 12 months, you will get an event with options to: a) Leave the metamorphosis form b) risk the demon within to take over your body (prowess dependant) (this gets progressively harder to contain the demon within if you choose to live in the metamorphosis form for way too long)
- If the demon within wins, you will get a notif event that asks you want to keep playing as the fallen demon hunter or as the player heir of the demon hunter (if they have an heir in the first place)
- Added a portrait modifier that let's you get horns when you become a demon hunter
- Added a portrait modifier that let's you get more demonic form by giving you horns and wings when in the metamorphosis state
- Normal Demon Hunters can take the decision to "Embrace the Demon" if they have atleast one of the Demon Hunter trait tracks maxxed out

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:
- To avoid conflicts with other branches in work, decision to become demon hunters or class related content/traits will now have their own files starting with `wc_class_`
### TODO:
- [x] Write descriptions for events that happen after taking the decision to "Embrace the Demon"
- [x] Find a way to let Demon Hunters have access to attain the metamorphosis form instead of blocking them out of it if they don't get the event where the demon within threatens them (`demon_hunter_events.0004` and `demon_hunter_events.0005`)

<!--
Before the merge, we recommend you to do these tests with your branch.
-->
## Tests:
- [x] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [x] The mod takes less than 5.5 GB in the Task Manager (Windows)

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:

FelModder committed:
First, Creating the Spectral Vision trait

FelModder committed:
Trait rebalancing and nerf since you aren't "Blind""

FelModder committed:
First Event and a new death reason for demon hunter ritual

FelModder committed:
2nd Event

FelModder committed:
Yeah this is whole DH flavour pack

FelModder committed:
fixes

FelModder committed:
moar fixes

FelModder committed:
Demon possession chance fixes

FelModder committed:
Metamorphosis trait and gain a blindfold if you get spectral vision

FelModder committed:
Embracing the decision

FelModder committed:
LOC fixes and general chance fixes

FelModder committed:
Setting up the last event and consequences

FelModder committed:
Trillion more events for DHs

FelModder committed:
Custom Tooltip for warning

FelModder committed:
some fixes

FelModder committed:
You become a Fallen Demon Hunter if you lose to Inner Demon

FelModder committed:
fallen demon hunter

FelModder committed:
fixes to some LOC

FelModder committed:
fixes and remove DH glow effect

FelModder committed:
.dds image by ryktruyur

FelModder committed:
Desc LOC fix

FelModder committed:
Portrait modifiers for Demon hunters and metamorphosis

FelModder committed:
Spectral Vision is health trait, not a fame

Grekopithikos committed:
blind and spectral are opposites

Grekopithikos committed:
Spectral Sight

Grekopithikos committed:
historical spectral

FelModder committed:
rename the trait dds

FelModder committed:
don't gatekeep metamorphosis behind the "bad" event

FelModder committed:
fixes

FelModder committed:
A bit broken commit

FelModder committed:
Endless fixes on this branch, fuck me...

FelModder committed:
Merge branch 'dev' into feat/code/spectral-vision-and-demon-hunter-event-chains

FelModder committed:
useless death reason

FelModder committed:
custom tooltip for metamorphosis requirements

FelModder committed:
Final Event

FelModder committed:
Temporary Scheme implementation

Dunno how to place and where to place an event that should launch when the scheme ends

FelModder committed:
Events without the localizations done

FelModder committed:
You need to be of the Iliidari faith

FelModder committed:
Error Log is code-hater pilled

Just solving some error log

FelModder committed:
scheme LOC complete

FelModder committed:
Final LOC event done

FelModder committed:
fixes fixes

FelModder committed:
Illidan can train you too

FelModder committed:
Merge branch 'dev' into feat/code/spectral-vision-and-demon-hunter-event-chains

FelModder committed:
Broken can also join Illidari

FelModder committed:
i'm so confused

FelModder committed:
someone end my suffering

FelModder committed:
re-adding life expectancy

FelModder committed:
I give up

FelModder committed:
forgor the odd_perspective

Grekopithikos committed:
Broken and Lost Ones cannot use Light magiv

FelModder committed:
interactions

Grekopithikos committed:
spelling mistakes

Grekopithikos committed:
is_demon_hunter_race_trigger

Grekopithikos committed:
random demon hunter horns

Grekopithikos commented on fonts/mapfont/Paradox_King_Script.otf:
Is this supposed to be a new mapfont?
Why in this branch?

FelModder replied:
Removed it. Fixed.

FelModder committed:
fix

TerrorBlades commented:
<img width="1018" height="768" alt="image" src="https://github.com/user-attachments/assets/42585154-4b81-4adf-98f7-66e2c7d85dd9" />
I have an option to make people "Become a Demon Hunter", I can do this to myself after I become a Demon Hunter. This seams to have no effect on anyone I use it on.

TerrorBlades commented:
<img width="1754" height="894" alt="image" src="https://github.com/user-attachments/assets/123daee3-bb00-478c-8d29-7598a150d338" />
Having no heir gives a error message. Oh and the first demon prey error

FelModder committed:
Remove everything related to the Scheme

Fuck that i guess...

FelModder committed:
Merge branch 'dev' into feat/code/spectral-vision-and-demon-hunter-event-chains

edmund-h commented:
> <img alt="image" width="1018" height="768" src="https://private-user-images.githubusercontent.com/105604501/487496352-42585154-4b81-4adf-98f7-66e2c7d85dd9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg1NzcwMDIsIm5iZiI6MTc1ODU3NjcwMiwicGF0aCI6Ii8xMDU2MDQ1MDEvNDg3NDk2MzUyLTQyNTg1MTU0LTRiODEtNGFkZi05OGY3LTY2ZTJjN2Q4NWRkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyMlQyMTMxNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZDU4NjA5NTUyOTYzZmNiZGY4ZDIzZTE0NjI1N2YxNzk5OWJhN2IxODcyY2U3NWFhZTQ3MzhmYjhmNGI0NDI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7UJamwZegCqBsOB46U2B3Wp1BGSGTUI-hCoQed5uqlk"> I have an option to make people "Become a Demon Hunter", I can do this to myself after I become a Demon Hunter. This seams to have no effect on anyone I use it on.

might not be related to this branch, but why does a draenei have a beard? that shouldnt be possible as they do not have facial hair

edmund-h commented on localization/english/event_localization/wc_demon_hunter_events_l_english.yml:
```diff
@@ -0,0 +1,62 @@
+
+ demon_hunter_0010_title:1 "A Destiny at End"
+ demon_hunter_0010_desc:0 "The [GetTrait('being_demon').GetName(GetNullCharacter)] contained within [new_fallen_hunter.GetFirstName] is no longer imprisoned. [first_demon_prey.GetFirstName]'s and [new_fallen_hunter.GetFirstName]'s souls have merged and any memory of either of their pasts is now gone. #help Will you play as the [GetTrait('fallen_demon_hunter').GetName(GetNullCharacter)]? Or as [pl ...
+ demon_hunter_0010_A:0 "Fate of the Fallen Hunter"
+ demon_hunter_0010_B:0 "Fate of the Hunter's Cub"
```
can we change these so that they are "in character" rather than just option labels? this feels very "game-y" in ways that CK3 events usually dont

FelModder replied:
I am changing and making a different description for when a character is alone and doesnt have an heir, this will be done in that.

But these options and this event itself will stay as is. `demon_hunter_0010` is *supposed* to be just a notifications event because i wasn't even gonna implement this event before.

But after the "sudden death", an event was needed as to what is happening/haven't to the player character.

edmund-h replied:
Can i just suggest a small change then? 
```suggestion
 demon_hunter_0010_A:0 "Choose the fate of the Fallen Hunter"
 demon_hunter_0010_B:0 "Choose the fate of the Hunter's Cub"
```

edmund-h commented on common/decisions/wc_class_traits_decisions.txt:
```diff
@@ -0,0 +1,250 @@
+﻿become_a_demon_hunter_decision = {
+	picture = { reference = "gfx/interface/illustrations/decisions/becoming_a_demon_hunter.dds" }
+	decision_group_type = major
```
this should not be a major decision. create a new decision group for the decisions in this file. call it something like class_trait_decision_group and localize it however you want. make the UI order priority whatever you feel is relevant

FelModder replied:
Good idea. Will make a different group for class decisions.

edmund-h commented on common/decisions/wc_class_traits_decisions.txt:
```diff
@@ -0,0 +1,250 @@
+
+embrace_the_demon_within = {
+	picture = { reference = "gfx/interface/illustrations/decisions/metamorphosis_form.dds" }
+	decision_group_type = major
```
should be added to new decision group, see above

edmund-h commented on common/decisions/wc_class_traits_decisions.txt:
```diff
@@ -0,0 +1,250 @@
+		}
+	}
+
+	cost = {
+		piety = 2000
+        prestige = 2000
+    }
```
these costs seem crazy high.

edmund-h commented on common/decisions/wc_class_traits_decisions.txt:
```diff
@@ -0,0 +1,250 @@
+		}
+	}
+
+	cost = { # Temporary cost for now, Change this to mana cost when the magic system comes around
+		prestige = 1000
+		piety = 2000
+	}
```
once again, these costs are kinda nutty

edmund-h commented on localization/english/event_localization/wc_demon_hunter_events_l_english.yml:
```diff
@@ -0,0 +1,62 @@
+ demon_hunter_0001_A:0 "...and then, I carve into its heart"
+
+ demon_hunter_0002_title:1 "A Destiny of Flame"
+ demon_hunter_0002_desc:0 "As the still beating heart of the [GetTrait('being_demon').GetName(GetNullCharacter)] still thumps in my hands, I had learnt that they pass away from whence they came... only to return to our worlds at will...\n\nBut I #italic know #!that they cannot come back if they never return to their world.\nI grab the heart and consume it with its filthy smell and energy and all, ...
```
this text needs improvement. here is my suggestion

```suggestion
  demon_hunter_0002_desc:0 "As the still-beating heart of the [GetTrait('being_demon').GetName(GetNullCharacter)] thumps in my hands, I realize how to stop the monster's cycle of rebirth in the [twisting_nether|E]:\nI consume the foul thing, energy and all, binding [first_demon_prey.GetFirstName] to my soul. As soon as I finish consuming the heart, I feel its dark energies fill my being.\n\n Suddenly, my surroundings change and my eyes burn as I try to grasp what I see: Numerous [GetTrait('being_demon').GetName(GetNullCharacter)]s marching onto countless worlds like ours, destroying them either through brute force or subtly from within.\n\nPlacid worlds reduced to ash and flame... I feel their pain resonate within my soul... I wish for the visions to stop but they keep going and going and going...\nAbandoning myself to my most desperate urge to stop them, I rip my own eyes clean from their sockets..."
```

FelModder replied:
I'll revisit this, thanks.

Grekopithikos committed:
Merge branch 'dev' into feat/code/spectral-vision-and-demon-hunter-event-chains

Grekopithikos committed:
dh triggers

FelModder committed:
Merge branch 'dev' into feat/code/spectral-vision-and-demon-hunter-event-chains

FelModder committed:
Make changes requested by Shatter

FelModder commented:
> <img alt="image" width="1018" height="768" src="https://private-user-images.githubusercontent.com/105604501/487496352-42585154-4b81-4adf-98f7-66e2c7d85dd9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk1NTY5ODEsIm5iZiI6MTc1OTU1NjY4MSwicGF0aCI6Ii8xMDU2MDQ1MDEvNDg3NDk2MzUyLTQyNTg1MTU0LTRiODEtNGFkZi05OGY3LTY2ZTJjN2Q4NWRkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAwNFQwNTQ0NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMDAzNzY3ZWI1NDdjMzQ4ZDVkNjE2OWUzNDFhYjgzYjk4NDEwNDQ0MjZkNjYxZGY5OTAwNDU2MjM5Yzk5NWE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OEvc9UMNtoZfWqckBRWOkNwtrr_RquJSeZ0YggOgI5E"> I have an option to make people "Become a Demon Hunter", I can do this to myself after I become a Demon Hunter. This seams to have no effect on anyone I use it on.

Removed the Scheme entirely. Fixed.

FelModder committed:
new decision_group_type and DH women will no longer be naked

FelModder commented:
> <img alt="image" width="1754" height="894" src="https://private-user-images.githubusercontent.com/105604501/487521158-123daee3-bb00-478c-8d29-7598a150d338.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk1NTg0ODMsIm5iZiI6MTc1OTU1ODE4MywicGF0aCI6Ii8xMDU2MDQ1MDEvNDg3NTIxMTU4LTEyM2RhZWUzLWJiMDAtNDc4Yy04ZDI5LTc1OThhMTUwZDMzOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAwNFQwNjA5NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MzlhNjhkZTNlZjk1NmEyOTQyY2ExZTY4OTQzM2MyMDMwN2EyZWUyMzc3MDhiNGZjNTU5MmIwMDQ1NjhkNGY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qAGb0529lihLO4ZiuiSS5NRQVT5ObrN1gupYUcVlN-U"> Having no heir gives a error message. Oh and the first demon prey error

Fixed

FelModder committed:
Desc error fixes

FelModder committed:
Update localization as suggested by Shatter

Co-authored-by: Edmund <447ght@gmail.com>

FelModder committed:
Grammer!

edmund-h reviewed (approved):
good work
